internal/graph/graph.go

```diff
@@ -137,16 +137,27 @@ func (g *Graph) InsertConstructor(ctor interface{}) error {
 	return nil
 }
 
-// ValidateReturnTypes validates if ctor's return type is already insterted in the graph
-func (g *Graph) ValidateReturnTypes(ctype reflect.Type) error {
+// ValidateReturnTypes validates if Invoke func's return type is Provided to the graph
+// checkCachedObjects=true ensures to throw an error only when the node is already
+// resolved and cached.
+//
+func (g *Graph) ValidateReturnTypes(ctype reflect.Type, checkCachedObjects bool) error {
```

Comment:
This function really confuses me now. I know that there is some code reuse going on, but we can we maybe split it into two separate ones?

Also, I feel like we should spell out, *exactly*, step-by-step what Invoke actually does because the code for everything involved in it no longer fits cleanly into my head.